backend/src/receipts.js

```diff
@@ -17,6 +18,41 @@ const INDEXER_COMPATIBILITY_RECEIPT_ACTION_KINDS = new Map([
   [null, "Unknown"],
 ]);
 
+const generateProperReceiptsList = async (receiptsList) => {
+  let receipts = [];
+  let actions;
+  let previousReceiptId = "";
+  const indexerCompatibilityTransactionActionKinds = await getIndexerCompatibilityTransactionActionKinds();
+  for (const receiptAction of receiptsList) {
+    if (previousReceiptId !== receiptAction.receipt_id) {
+      previousReceiptId = receiptAction.receipt_id;
+      actions = [];
+      const receipt = {
+        actions,
+        blockTimestamp: new BN(receiptAction.executed_in_block_timestamp)
+          .divn(10 ** 6)
+          .toNumber(),
+        gasBurnt: receiptAction.gas_burnt,
+        receiptId: receiptAction.receipt_id,
+        receiverId: receiptAction.receiver_id,
+        signerId: receiptAction.predecessor_id,
+        status: INDEXER_COMPATIBILITY_RECEIPT_ACTION_KINDS.get(
+          receiptAction.status
+        ),
+        originatedFromTransactionHash:
+          receiptAction.originated_from_transaction_hash,
+        tokensBurnt: receiptAction.tokens_burnt,
+      };
+      receipts.push(receipt);
+    }
+    actions.push({
+      args: receiptAction.args,
+      kind: indexerCompatibilityTransactionActionKinds.get(receiptAction.kind),
+    });
+  }
+  return receipts;
+};
+
```

Comment:
This part is just for demo and if we'll leave this implementation I'll refactor it